pkg/internal/token/options.go

```diff
@@ -63,10 +63,16 @@ func GetSupportedLogins() string {
 }
 
 func NewOptions() Options {
+	env_TokenCacheDir := os.Getenv("KUBECACHEDIR")
 	return Options{
-		LoginMethod:   DeviceCodeLogin,
-		Environment:   defaultEnvironmentName,
-		TokenCacheDir: DefaultTokenCacheDir,
+		LoginMethod: DeviceCodeLogin,
+		Environment: defaultEnvironmentName,
+		TokenCacheDir: func() string {
```

Comment:
you can just check if `env_TokenCacheDir` is empty or not right after L66 and set it accordingly.

Reply:
does the change follow what you had in mind @weinong ?

if yes, I'll resolver the conversation.
thank you for your advice